sjsonnet/src/sjsonnet/Std.scala

```diff
@@ -788,39 +788,39 @@ class Std(private val additionalNativeFunctions: Map[String, Val.Builtin] = Map.
   }
 
   private object Range extends Val.Builtin2("range", "from", "to") {
-    def evalRhs(from: Val, to: Val, ev: EvalScope, pos: Position): Val =
+    def evalRhs(from: Lazy, to: Lazy, ev: EvalScope, pos: Position): Val =
     new Val.Arr(
       pos,
-      (from.asInt to to.asInt).map(i => Val.Num(pos, i)).toArray
+      (from.force.asInt to to.force.asInt).map(i => Val.Num(pos, i)).toArray
     )
   }
 
   private object ManifestJson extends Val.Builtin1("manifestJson", "v") {
-    def evalRhs(v: Val, ev: EvalScope, pos: Position): Val =
-      Val.Str(pos, Materializer.apply0(v, MaterializeJsonRenderer())(ev).toString)
+    def evalRhs(v: Lazy, ev: EvalScope, pos: Position): Val =
+      Val.Str(pos, Materializer.apply0(v.force, MaterializeJsonRenderer())(ev).toString)
   }
 
   private object ManifestJsonMinified extends Val.Builtin1("manifestJsonMinified", "v") {
-    def evalRhs(v: Val, ev: EvalScope, pos: Position): Val =
-      Val.Str(pos, Materializer.apply0(v, new MaterializeJsonRenderer(indent = -1, newline = "", keyValueSeparator = ":"))(ev).toString)
+    def evalRhs(v: Lazy, ev: EvalScope, pos: Position): Val =
+      Val.Str(pos, Materializer.apply0(v.force, new MaterializeJsonRenderer(indent = -1, newline = "", keyValueSeparator = ":"))(ev).toString)
   }
 
   private object ManifestJsonEx extends Val.Builtin4("manifestJsonEx", "value", "indent", "newline", "key_val_sep", Array(null, null, Val.Str(dummyPos, "\n"), Val.Str(dummyPos, ": "))) {
-    def evalRhs(v: Val, i: Val, newline: Val, keyValSep: Val, ev: EvalScope, pos: Position): Val =
+    def evalRhs(v: Lazy, i: Lazy, newline: Lazy, keyValSep: Lazy, ev: EvalScope, pos: Position): Val =
       Val.Str(pos, Materializer
-        .apply0(v, MaterializeJsonRenderer(indent = i.asString.length, newline = newline.asString, keyValueSeparator = keyValSep.asString))(ev)
+        .apply0(v.force, MaterializeJsonRenderer(indent = i.force.asString.length, newline = newline.force.asString, keyValueSeparator = keyValSep.force.asString))(ev)
         .toString)
   }
 
   private object ParseJson extends Val.Builtin1("parseJson", "str") {
-    def evalRhs(str: Val, ev: EvalScope, pos: Position): Val =
-      ujson.StringParser.transform(str.asString, new ValVisitor(pos))
+    def evalRhs(str: Lazy, ev: EvalScope, pos: Position): Val =
+      ujson.StringParser.transform(str.force.asString, new ValVisitor(pos))
   }
 
   private object ParseYaml extends Val.Builtin1("parseYaml", "str") {
-    def evalRhs(str: Val, ev: EvalScope, pos: Position): Val = {
+    def evalRhs(str: Lazy, ev: EvalScope, pos: Position): Val = {
       try {
-        ujson.StringParser.transform(Platform.yamlToJson(str.asString), new ValVisitor(pos))
+        ujson.StringParser.transform(Platform.yamlToJson(str.force.asString), new ValVisitor(pos))
       } catch {
         case _: Exception => null
       }
```

Comment:
Pre-existing bug in nearby code (below the fold):

```
private object ManifestTomlEx extends Val.Builtin2("manifestTomlEx", "value", "indent") {
    private def isTableArray(v: Val) = v match {
      case s: Val.Arr => s.length > 0 && s.asLazyArray.forall(_.isInstanceOf[Val.Obj])
      case _ => false
    }
```

I think we need a `_.force.isInstanceOf` instead of what we have now, i.e. add the force.